cdap-ui/app/cdap/components/AbstractWidget/HierarchyWidget/dataHandler.ts

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { IStageSchema } from 'components/AbstractWidget';
+import uuidV4 from 'uuid/v4';
+
```

Comment:
There are too many helper functions in this module that converts the types from a JSON to a flat array and back to JSON. 
- Could you please add types to the input and output for these functions
- Add comments on what the function is doing that way we have a high level overview on how the data structure is managed and how it is changed.